crates/bevy_asset/src/path.rs

```diff
@@ -465,6 +465,38 @@ impl<'a> AssetPath<'a> {
         Some(extension)
     }
 
+    /// Returns the path of the [`AssetPath`] relative to the app root.
+    /// Returns None if the [`AssetPath`] uses a non-default Asset Source.
+    ///
+    /// Ex: an [`AssetPath`] referencing "/models/apple.gltf" would return a [`PathBuf`] for "assets/models/apple.gltf"
+    pub fn get_root_relative_path(&self) -> Option<PathBuf> {
```

Comment:
I see what you mean, it's a tricky name. On balance, I think you could get away with `get_relative_path` because the project root is implied, but not sure TBH.